src/contact/contact.controller.ts

```diff
@@ -0,0 +1,64 @@
+import { Response } from 'express';
+import { Context } from '../middleware/context';
+import { UserContactRepository } from './user-contact.repository';
+import { Repository } from '../service/repository';
+import { CreateContactDto } from './dto/create-contact.dto';
+
+const repo = new Repository<UserContactRepository>(UserContactRepository);
+
+const getRepoFromUsername = (req: Context): UserContactRepository => {
+  const username = req.context?.username || '';
```

Comment:
Nit pick: Should this be needed? In my mind, it would be better to make the middleware fail if a username isn't supplied. So `req.context.username` should always exist